lib/hammer_cli_katello/content_view_version.rb

```diff
@@ -412,19 +428,44 @@ def execute
         untar_export
 
         export_json = read_json
-
         cv = content_view(export_json['name'], options['option_organization_id'])
-        sync_repositories(export_json['repositories'], options['option_organization_id'])
-
-        publish(
-          cv['id'],
-          export_json['major'],
-          export_json['minor'],
-          repos_units(export_json['repositories'])
-        )
+
+        if export_json['composite_components']
+          composite_version_ids = export_json['composite_components'].map {|component| find_local_component_id(component) }
+
+          resource_content_views.call(:update,
+                                      'id' => cv['id'], 'component_ids' => composite_version_ids)
+          publish(cv['id'], export_json['major'], export_json['minor'], nil)
+        else
+          sync_repositories(export_json['repositories'], options['option_organization_id'])
+
+          publish(
+            cv['id'],
+            export_json['major'],
+            export_json['minor'],
+            repos_units(export_json['repositories'])
+          )
+        end
         return HammerCLI::EX_OK
       end
 
+      def find_local_component_id(component_from_export)
+        *name, version = component_from_export.split(' ')
```

Comment:
What does this use of `*` and two return parameters result in? An example would be helpful to wrap my brain around this bit of code.

Reply:
Basically it says "put everything but the last thing in the array into `name`, and the last thing in `version`. For example:

```ruby
irb(main):001:0> *foo, bar = [1,2,3,4,5]
=> [1, 2, 3, 4, 5]
irb(main):002:0> bar
=> 5
irb(main):003:0> foo
=> [1, 2, 3, 4]
```
When we fetch the components from the CV, they are in "name + version number" format, and I want to split the name (which may have spaces) off from the version.